### PR TITLE
Esp32 s3

### LINKED ISSUE #27: ESP32S3 support
as is the [master_simple.ino](https://github.com/hideakitai/ESP32DMASPI/tree/master/examples/master_simple) example does not compile:
```
In file included from /home/stefan/mydata/arduino_sketchbook/libraries/ESP32DMASPI/examples/master_simple/master_simple.ino:1:
/home/stefan/mydata/arduino_sketchbook/libraries/ESP32DMASPI/ESP32DMASPIMaster.h:51:5: warning: missing initializer for member 'spi_bus_config_t::<anonymous>' [-Wmissing-field-initializers]
     };
     ^
/home/stefan/mydata/arduino_sketchbook/libraries/ESP32DMASPI/ESP32DMASPIMaster.h:51:5: warning: missing initializer for member 'spi_bus_config_t::<anonymous>' [-Wmissing-field-initializers]
/home/stefan/mydata/arduino_sketchbook/libraries/ESP32DMASPI/ESP32DMASPIMaster.h:51:5: warning: missing initializer for member 'spi_bus_config_t::data4_io_num' [-Wmissing-field-initializers]
/home/stefan/mydata/arduino_sketchbook/libraries/ESP32DMASPI/ESP32DMASPIMaster.h:51:5: warning: missing initializer for member 'spi_bus_config_t::data5_io_num' [-Wmissing-field-initializers]
/home/stefan/mydata/arduino_sketchbook/libraries/ESP32DMASPI/ESP32DMASPIMaster.h:51:5: warning: missing initializer for member 'spi_bus_config_t::data6_io_num' [-Wmissing-field-initializers]
/home/stefan/mydata/arduino_sketchbook/libraries/ESP32DMASPI/ESP32DMASPIMaster.h:51:5: warning: missing initializer for member 'spi_bus_config_t::data7_io_num' [-Wmissing-field-initializers]
/home/stefan/mydata/arduino_sketchbook/libraries/ESP32DMASPI/ESP32DMASPIMaster.h:51:5: warning: missing initializer for member 'spi_bus_config_t::intr_flags' [-Wmissing-field-initializers]
/home/stefan/mydata/arduino_sketchbook/libraries/ESP32DMASPI/ESP32DMASPIMaster.h:53:29: error: 'HSPI_HOST' was not declared in this scope
     spi_host_device_t host {HSPI_HOST};
                             ^~~~~~~~~
/home/stefan/mydata/arduino_sketchbook/libraries/ESP32DMASPI/ESP32DMASPIMaster.h:53:29: note: suggested alternative: 'SPI3_HOST'
     spi_host_device_t host {HSPI_HOST};
                             ^~~~~~~~~
                             SPI3_HOST
/home/stefan/mydata/arduino_sketchbook/libraries/ESP32DMASPI/ESP32DMASPIMaster.h:53:38: error: cannot convert '<brace-enclosed initializer list>' to 'spi_host_device_t' in initialization
     spi_host_device_t host {HSPI_HOST};
                                      ^
/home/stefan/mydata/arduino_sketchbook/libraries/ESP32DMASPI/ESP32DMASPIMaster.h: In member function 'bool arduino::esp32::spi::dma::Master::begin(uint8_t)':
/home/stefan/mydata/arduino_sketchbook/libraries/ESP32DMASPI/ESP32DMASPIMaster.h:61:43: error: 'VSPI' was not declared in this scope
         bus_cfg.mosi_io_num = (spi_bus == VSPI) ? MOSI : 13;
                                           ^~~~
/home/stefan/mydata/arduino_sketchbook/libraries/ESP32DMASPI/ESP32DMASPIMaster.h:61:43: note: suggested alternative: 'SPI'
         bus_cfg.mosi_io_num = (spi_bus == VSPI) ? MOSI : 13;
                                           ^~~~
                                           SPI
/home/stefan/mydata/arduino_sketchbook/libraries/ESP32DMASPI/ESP32DMASPIMaster.h: In member function 'bool arduino::esp32::spi::dma::Master::initialize(uint8_t)':
/home/stefan/mydata/arduino_sketchbook/libraries/ESP32DMASPI/ESP32DMASPIMaster.h:297:36: error: 'HSPI_HOST' was not declared in this scope
         host = (spi_bus == HSPI) ? HSPI_HOST : VSPI_HOST;
                                    ^~~~~~~~~
/home/stefan/mydata/arduino_sketchbook/libraries/ESP32DMASPI/ESP32DMASPIMaster.h:297:36: note: suggested alternative: 'SPI3_HOST'
         host = (spi_bus == HSPI) ? HSPI_HOST : VSPI_HOST;
                                    ^~~~~~~~~
                                    SPI3_HOST
/home/stefan/mydata/arduino_sketchbook/libraries/ESP32DMASPI/ESP32DMASPIMaster.h:297:48: error: 'VSPI_HOST' was not declared in this scope
         host = (spi_bus == HSPI) ? HSPI_HOST : VSPI_HOST;
                                                ^~~~~~~~~
/home/stefan/mydata/arduino_sketchbook/libraries/ESP32DMASPI/ESP32DMASPIMaster.h:297:48: note: suggested alternative: 'SPI3_HOST'
         host = (spi_bus == HSPI) ? HSPI_HOST : VSPI_HOST;
                                                ^~~~~~~~~
                                                SPI3_HOST
Using library ESP32DMASPI at version 0.2.0 in folder: /home/stefan/mydata/arduino_sketchbook/libraries/ESP32DMASPI 
Using library SPI at version 2.0.0 in folder: /home/stefan/.arduino15/packages/esp32/hardware/esp32/2.0.6/libraries/SPI 
exit status 1
Error compiling for board ESP32S3 Dev Module.
```

i will test if the #25 fixes this..

### PR DESCRIPTION
**Pull request by s-light**
based on #25 
fixes #27

### CODE REVIEW

File: examples/master_simple/debug.cfg
Comment (hideakitai) [inline examples/master_simple/debug.cfg:1]: Could you remove unnecessary files for this library?

File: examples/master_simple/debug_custom.json
Comment (hideakitai) [inline examples/master_simple/debug_custom.json:1]: Could you remove unnecessary files for this library?

File: examples/master_simple/esp32s3.svd
Comment (hideakitai) [inline examples/master_simple/esp32s3.svd:1]: Could you remove unnecessary files for this library?

File: ESP32DMASPIMaster.h [hunk 1]
```diff
@@ -22,6 +22,28 @@
     }
 #endif
 
+#if CONFIG_IDF_TARGET_ESP32S2 || CONFIG_IDF_TARGET_ESP32S3
+    #define VSPI FSPI
+    #define VSPI_HOST FSPI_HOST
+#endif
+
+// TODO check ESP32C3
```
Comment (hideakitai) [inline ESP32DMASPIMaster.h:30]: Could you also support ESP32C3?

**Review by hideakitai — CHANGES_REQUESTED** [review]

**Comment by hideakitai** [comment]
Fixed by https://github.com/hideakitai/ESP32DMASPI/pull/31

**Comment by s-light** [comment]
thanks for making it work.
sorry for not seeing your requests.

**Pull request closed without merging**